options/options_test.cc

```diff
@@ -529,6 +531,65 @@ TEST_F(OptionsTest, GetBlockBasedTableOptionsFromString) {
   ASSERT_EQ(table_opt.cache_index_and_filter_blocks,
             new_opt.cache_index_and_filter_blocks);
   ASSERT_EQ(table_opt.filter_policy, new_opt.filter_policy);
+
+  // make sure default values of block cache are overwritten
+  ASSERT_OK(GetBlockBasedTableOptionsFromString(table_opt,
+             "cache_index_and_filter_blocks=1;"
+             "block_cache=1M:4:true:0.5;"
```

Comment:
We have a new pattern for (de)serializing structs in a more consistent/structured way. You can look at how "compaction_options_fifo" is implemented using `ParseStructOptions()`. Do you want to try it out?

Reply:
I updated the PR to use ParseStructOptions. Here are the changes in the commit:
1) Created a LRUCacheOptions structure to hold the required cache
options.
2) A new function NewLRUCache now accepts LRUCacheOptions to create the
cache. This in turn calls the previous NewLRUCache function.
3) lru_cache_options_type_info is added to option_helper.h. This has
type info required for ParseStructOptions.
4) A new option type, kLRUCacheOptions is added and ParseOptionHelper is
modified to parse this new type using ParseStructOptions.
5) ParseBlockBasedTableOption is modified to support block_cache and
block_cache_compressed options in struct like format as
block_cache={capacity=1M;num_shard_bits=4;strict_capacity_limit=true;high_pri_pool_ratio=0.5;}
Backward compatibility is also ensured.
6) Appropriate unit tests are added.